File: apps/api/src/constants/provinces.ts
// Ontario FSA prefixes — used for province detection from postal code
// MVP: Ontario only. K, L, M, N, P = Ontario.
export const ONTARIO_FSA_PREFIXES = ['K', 'L', 'M', 'N', 'P'] as const

export function isOntarioPostalCode(postalCode: string): boolean {
  const fsa = postalCode.trim().toUpperCase().charAt(0)
  return (ONTARIO_FSA_PREFIXES as readonly string[]).includes(fsa)
}

// Ontario Land Transfer Tax brackets
// Source: ontario.ca — updated May 2026
export const ONTARIO_LTT_BRACKETS = [
  { upTo: 55_000,    rate: 0.005 },
  { upTo: 250_000,   rate: 0.010 },
  { upTo: 400_000,   rate: 0.015 },
  { upTo: 2_000_000, rate: 0.020 },
  { upTo: Infinity,  rate: 0.025 },  // 2.5% on amounts above $2M
] as const

// Toronto Municipal Land Transfer Tax brackets (applies on top of provincial)
export const TORONTO_MLTT_BRACKETS = [
  { upTo: 55_000,    rate: 0.005 },
  { upTo: 400_000,   rate: 0.010 },
  { upTo: 2_000_000, rate: 0.020 },
  { upTo: Infinity,  rate: 0.025 },
] as const

File: services/calc-engine/constants/provinces.py
"""
Province-specific data: Ontario FSA prefixes and LTT brackets.
MVP covers Ontario only.
"""

# Ontario FSA prefixes — first letter of postal code
# K, L, M, N, P = Ontario
ONTARIO_FSA_PREFIXES: frozenset[str] = frozenset({'K', 'L', 'M', 'N', 'P'})


def is_ontario_postal_code(postal_code: str) -> bool:
    """
    Return True if the postal code belongs to Ontario.

    Args:
        postal_code: Canadian postal code (e.g. "L4J 7K1").

    Returns:
        True if the FSA prefix is one of K, L, M, N, P.
    """
    fsa = postal_code.strip().upper()[0] if postal_code.strip() else ''
    return fsa in ONTARIO_FSA_PREFIXES


# Ontario Land Transfer Tax brackets (provincial)
# Source: ontario.ca — updated May 2026
ONTARIO_LTT_BRACKETS: list[tuple[float, float]] = [
    (55_000,      0.005),
    (250_000,     0.010),
    (400_000,     0.015),
    (2_000_000,   0.020),
    (float('inf'), 0.025),
]

# Toronto Municipal Land Transfer Tax brackets (applies on top of provincial)
TORONTO_MLTT_BRACKETS: list[tuple[float, float]] = [
    (55_000,      0.005),
    (400_000,     0.010),
    (2_000_000,   0.020),
    (float('inf'), 0.025),
]
